### PR TITLE
Add ClientFactory.numConnections()

### LINKED ISSUE #3596: Add `ClientFactory.numConnections()`
.. like `Server` has `numConnections()`, so that a user can easily figure out how many connections it is managing.

### PR DESCRIPTION
**Pull request by kojilin**
Motivation:

It's be nice if `ClientFactory` has `numConnections()` like `Server`,
so that a user can easily figure out how many connections it is managing.

Modifications:

- Add `Clientfactory.numConnections()`, get connection count from allChannels directly.
- Add test for multiple h1c request & multie servers

Result:

- Closes #3596

### CODE REVIEW

File: core/src/main/java/com/linecorp/armeria/client/ClientFactory.java [hunk 1]
```diff
@@ -196,6 +196,11 @@ static void disableShutdownHook() {
      */
     Object newClient(ClientBuilderParams params);
 
+    /**
+     * Returns the number of open connections on this {@link ClientFactory}.
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/client/ClientFactory.java:200]:
```suggestion
     * Returns the number of open connections managed by this {@link ClientFactory}.
```

**Review by trustin — COMMENTED** [review]
Thank you so much for working on this, @kojilin :bow:

**Review by trustin — APPROVED** [review]

File: core/src/test/java/com/linecorp/armeria/client/HttpClientFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+class HttpClientFactoryTest {
+    @RegisterExtension
+    public static final ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            sb.service("/", new AbstractHttpService() {
+                @Override
+                protected HttpResponse doGet(ServiceRequestContext ctx, HttpRequest req) throws Exception {
+                    return HttpResponse.from(new CompletableFuture<>());
```
Comment (ikhoon) [inline core/src/test/java/com/linecorp/armeria/client/HttpClientFactoryTest.java:44]: nit: `HttpResponse.streaming()`? or you can leave it as it is.

**Review by ikhoon — APPROVED** [review]
Thanks a lot! @kojilin

**Review by minwoox — APPROVED** [review]
👍 👍 👍

**Pull request merged**